changes/2020-07-14_multi-keyring-require-generation/change.md

```diff
@@ -0,0 +1,108 @@
+[//]: # "Copyright Amazon.com Inc. or its affiliates. All Rights Reserved."
+[//]: # "SPDX-License-Identifier: CC-BY-SA-4.0"
+
+# Require Generation in the Multi-Keyring
+
+## Affected Features
+
+This serves as a reference of all features that this change affects.
+
+| Feature                                                                                           |
+| ------------------------------------------------------------------------------------------------- |
+| [Multi-Keyring](../../framework/multi-keyring.md)                                                 |
+| [Multi keyring generator](https://github.com/awslabs/aws-encryption-sdk-specification/issues/114) |
+
+## Affected Specifications
+
+This serves as a reference of all specification documents that this change affects.
+
+| Specification                                     |
+| ------------------------------------------------- |
+| [Multi-Keyring](../../framework/multi-keyring.md) |
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+This change additionally affects the unreleased AWS KMS keyring implementations for Java and Python.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The multi-keyring has a clearly configured intent of which keyring SHOULD be used to generate the data key,
+but this is not actually enforced during the encryption operation.
+This change strengthens the requirements on multi-keyrings such that if a generator is configured,
+it MUST generate the data key.
+
+## Out of Scope
+
+Detecting at keyring configuration time that encryption will always fail
+because keyrings are not capable of meeting these requirements
```

Comment:
Might be worth saying what "these" requirements are. Is it "because the generator keyring cannot call AWS KMS GenerateDataKey or already has a plaintext data key" or is there something more?

Reply:
Left over from an earlier version that had at least one more requirement (that turned out to be already specified). But the general concern is still more broad so I'll indicate that.